temporalcli/commands.workflow_exec.go

```diff
@@ -194,16 +195,191 @@ func (c *TemporalWorkflowSignalWithStartCommand) run(cctx *CommandContext, _ []s
 		RunId      string `json:"runId"`
 		Type       string `json:"type"`
 		Namespace  string `json:"namespace"`
-		TaskQueue  string `json:"taskQueue"`
 	}{
 		WorkflowId: c.WorkflowId,
 		RunId:      resp.RunId,
 		Type:       c.Type,
 		Namespace:  c.Parent.Namespace,
-		TaskQueue:  c.TaskQueue,
 	}, printer.StructuredOptions{})
 }
 
+func (c *TemporalWorkflowStartUpdateWithStartCommand) run(cctx *CommandContext, _ []string) error {
+	waitForStage := client.WorkflowUpdateStageUnspecified
+	switch c.UpdateWaitForStage.Value {
+	case "accepted":
+		waitForStage = client.WorkflowUpdateStageAccepted
+	}
+	if waitForStage != client.WorkflowUpdateStageAccepted {
+		return fmt.Errorf("invalid wait for stage: %v, valid values are: 'accepted'", c.UpdateWaitForStage)
+	}
+
+	updatePayloadInputOpts := PayloadInputOptions{
+		Input:       c.UpdateInput,
+		InputFile:   c.UpdateInputFile,
+		InputMeta:   c.UpdateInputMeta,
+		InputBase64: c.UpdateInputBase64,
+	}
+	updateInput, err := updatePayloadInputOpts.buildRawInput()
+	if err != nil {
+		return err
+	}
+	updateOpts := client.UpdateWorkflowOptions{
+		UpdateID:            c.UpdateId,
+		WorkflowID:          c.WorkflowId,
+		RunID:               c.RunId,
+		UpdateName:          c.UpdateName,
+		Args:                updateInput,
+		WaitForStage:        waitForStage,
+		FirstExecutionRunID: c.UpdateFirstExecutionRunId,
+	}
+
+	handle, err := executeUpdateWithStartWorkflow(
+		cctx,
+		c.Parent.ClientOptions,
+		c.SharedWorkflowStartOptions,
+		c.WorkflowStartOptions,
+		c.PayloadInputOptions,
+		updateOpts,
+	)
+	if err != nil {
+		return err
+	}
+
+	// Currently we only accept 'accepted' as a valid wait for stage value, but we intend
+	// to support more in the future.
+	if waitForStage == client.WorkflowUpdateStageAccepted {
+		// Use a canceled context to check whether the initial server response
+		// shows that the update has _already_ failed, without issuing a second request.
+		ctx, cancel := context.WithCancel(cctx)
+		cancel()
+		err = handle.Get(ctx, nil)
+		var timeoutOrCanceledErr *client.WorkflowUpdateServiceTimeoutOrCanceledError
+		if err != nil && !errors.As(err, &timeoutOrCanceledErr) {
+			return fmt.Errorf("unable to update workflow: %w", err)
+		}
+	}
```

Comment:
Hrmm, I don't think `UpdateWithStartWorkflow` will return until the wait stage is reached, and if there is some error, it will be reflected in the error to that call. @dandavison - is this correct?

Reply:
An example is sending a request for an update name that does not exist. `err` is `nil`, but the update is not accepted, and `handle` contains within it an error, hence this code (which is also present in CLI update impl) to extract that error without making a network request.

Reply:
Ah, I see, there is "accepted but failed". We should consider adding a `Done() bool` to `WorkflowUpdateHandle` for this, but in the meantime, all good here.